Frends.AzureDataLake.DeleteContainer/Frends.AzureDataLake.DeleteContainer/DeleteContainer.cs

```diff
@@ -0,0 +1,103 @@
+    /// <returns>Object { string ContainerWasDeleted, string Message }</returns>
+    public static async Task<Result> DeleteContainer(
+        [PropertyTab] Input input,
+        [PropertyTab] Options options,
+        CancellationToken cancellationToken
+    )
+    {
+        if (
+            input.ConnectionMethod is ConnectionMethod.OAuth2
+            && (
+                input.ApplicationID is null
+                || input.ClientSecret is null
+                || input.TenantID is null
+                || input.StorageAccountName is null
+            )
+        )
+            throw new InvalidInputException(
+                "Input.StorageAccountName, Input.ClientSecret, Input.ApplicationID and Input.TenantID parameters can't be empty when Input.ConnectionMethod = OAuth."
+            );
+        if (
+            input.ConnectionMethod is ConnectionMethod.ConnectionString
+            && string.IsNullOrWhiteSpace(input.ConnectionString)
+        )
+            throw new InvalidInputException(
+                "ConnectionString parameter can't be empty when Input.ConnectionMethod = ConnectionString."
+            );
+        if (string.IsNullOrWhiteSpace(input.ContainerName))
+            throw new InvalidInputException("ContainerName parameter can't be empty.");
+
+        var container = GetDataLakeContainer(input);
+        try
+        {
+            var result = await container.DeleteIfExistsAsync(null, cancellationToken);
+            //can throw if wrong secrets etc
+
+            if (!result)
+            {
+                if (options.ThrowErrorIfContainerDoesNotExists)
+                    throw new ContainerNotFoundException();
+                else
+                    return new Result(false, "Container not found.");
+            }
+            return new Result(result, "Container deleted successfully.");
+        }
+        catch (AuthenticationFailedException)
+        {
+            throw;
+        }
+    }
+
+    private static DataLakeFileSystemClient GetDataLakeContainer(Input input)
+    {
+        try
+        {
+            DataLakeServiceClient client;
+
+            if (input.ConnectionMethod is ConnectionMethod.ConnectionString)
+                client = new DataLakeServiceClient(input.ConnectionString);
+            else
+            {
+                var credentials = new ClientSecretCredential(
+                    input.TenantID,
+                    input.ApplicationID,
+                    input.ClientSecret,
+                    new ClientSecretCredentialOptions()
+                );
+                client = new DataLakeServiceClient(
+                    new Uri($"https://{input.StorageAccountName}.dfs.core.windows.net"),
+                    credentials
+                );
+            }
+
+            return client.GetFileSystemClient(input.ContainerName);
+        }
+        catch (FormatException ex)
+        {
+            throw new FormatException($"GetDataLakeContainer error {ex}");
```

Comment:
You should encapsulate the previous exception as inner exception. And overall it seems that you want to provide additional context to any exception (not just format exception). So maybe catch "Exception ex" and retrhow as inner exception `throw new Exception("your message", ex)`. This way the new exception will still contain all the info on the inner exception.